great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+from typing import Dict, List, Optional, Union
+
+from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
+from great_expectations.exceptions import ProfilerConfigurationError
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.domain_builder import DomainBuilder
+from great_expectations.rule_based_profiler.domain_builder.cardinality_checker import (
+    CardinalityChecker,
+    CardinalityLimitMode,
+)
+from great_expectations.rule_based_profiler.domain_builder.domain_builder import (
+    build_domains_from_column_names,
+)
+from great_expectations.rule_based_profiler.types import Domain, ParameterContainer
+from great_expectations.validator.metric_configuration import MetricConfiguration
+from great_expectations.validator.validator import Validator
+
+
+class CategoricalColumnDomainBuilder(DomainBuilder):
+    """
+    This DomainBuilder uses column cardinality to identify domains.
+    """
+
+    def __init__(
+        self,
+        data_context: "DataContext",  # noqa: F821
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+        limit_mode: Optional[Union[CardinalityLimitMode, str]] = None,
+        max_unique_values: Optional[int] = None,
+        max_proportion_unique: Optional[int] = None,
+        exclude_columns: Optional[List[str]] = None,
+    ):
+        """Create column domains where cardinality is within the specified limit.
+
+        Cardinality refers to the number of unique values in a given domain.
+        Categorical generally refers to columns with relatively limited
+        number of unique values.
+        Limit mode can be absolute (number of unique values) or relative
+        (proportion of unique values). You can choose one of: limit_mode,
+        max_unique_values or max_proportion_unique to specify the cardinality
+        limit.
+        Note that the limit must be met for each batch separately that is
```

Comment:
@anthonyburdi In addition to satisfying cardinality requirements/limits at a `Batch` level, do we also have aggregated across all `Batch` objects requirements for cardinality (i.e., no more than X for all batches)?  Thanks.

Reply:
Not currently, checking with Eugene on requirements. Will report back.

Reply:
In the current PR we do not, but in the future we can further extend CardinalityLimit with additional subclasses that handle appearance and recency thresholds e.g. cardinality limit checked against "num of unique values that appear in x% of batches" and "num of unique values that appear in the latest 20% of batches"

Reply:
@anthonyburdi It would be wonderful (as a nice-to-have) to document future possibility, such as this, succinctly, in the code.  Thanks!